src/allotropy/parsers/ctl_immunospot/ctl_immunospot_structure.py

```diff
@@ -0,0 +1,243 @@
+        raw_name = assert_not_none(
+            reader.pop(),
+            msg="Unable to read name of assay data.",
+        )
+
+        name_match = assert_not_none(
+            re.search("\t\t\t(.+)", raw_name),
+            msg="Unable to parse name of assay data.",
+        )
+        name = re.sub(r"\s+", " ", name_match.group(1))
+
+        if "ImmunoSpot Plate Code" in name:
+            name = "Spot Count"
+
+        raw_columns = assert_not_none(
+            reader.pop(),
+            msg=f"Unable to read data from {name}",
+        )
+        columns = re.sub(r"\s+", " ", raw_columns.strip())
+
+        reader.drop_empty()
+
+        wells = {}
+        for raw_line in reader.pop_until_empty():
+            line = re.sub(r"\s+", " ", raw_line)
+            if match := re.match(r"^([A-Z])\s+(.+)", line):
+                raw_values = match.group(2).strip()
+                for column, value in zip(columns.split(), raw_values.split()):
+                    well = Well(
+                        col=column,
+                        row=match.group(1),
+                        value=try_float_or_none(value),
+                    )
+                    wells[well.pos] = well
+
+        return Plate(name, wells)
+
+    def get_well_count(self) -> int:
+        return len(self.wells)
+
+    def iter_wells(self) -> Iterator[Well]:
+        yield from self.wells.values()
+
+    def get_well(self, pos: str) -> Well:
+        return assert_not_none(
+            self.wells.get(pos),
+            msg=f"Unable to find well '{pos}' in plate '{self.name}'",
+        )
+
+
+@dataclass
+class AssayData:
+    plates: list[Plate]
+    identifier: Optional[str]
+    well_count: int
+
+    @staticmethod
+    def create(reader: LinesReader) -> AssayData:
+        reader.drop_until_inclusive("Unprocessed Data$")
+
+        reader.drop_empty()
+        plate_code_line = reader.get() or ""
+        assert_not_none(
+            re.search("Plate Code =", plate_code_line),
+            msg="Unable to find ImmunoSpot Plate Code line",
+        )
+
+        identifier = (
+            match.group(1)
+            if (match := re.search(r"Plate Code = ([\w ]+)", plate_code_line))
+            else None
+        )
+
+        plates = []
+        while reader.current_line_exists():
+            reader.drop_empty()
+            plates.append(Plate.create(reader))
+            reader.drop_empty()
+
+        if not plates:
+            error = "Unable to find plate information."
+            raise AllotropeConversionError(error)
+
+        return AssayData(plates, identifier, well_count=plates[0].get_well_count())
+
+    def iter_wells(self) -> Iterator[Well]:
+        yield from self.plates[0].iter_wells()
+
+    def iter_plates_well(self, pos: str) -> Iterator[Well]:
+        for plate in self.plates:
+            yield plate.get_well(pos)
+
+
+@dataclass
+class Data:
+    device_info: DeviceInfo
+    assay_info: AssayInfo
```

Comment:
This doesn't appear to be used anywhere, and you're not really doing any parsing of the lines?

If this will be used in a follow-up PR, then you can just add a comment here saying use will be implemented soon. 

If it is not used, you should remove it, otherwise, you should just skip over it in the create method by calling the `drop_until_empty` method with a comment about skipping over these lines.